src/Navigation.js

```diff
@@ -51,6 +55,31 @@ class Navigation extends React.Component {
       </div>
     );
   }
+
```

Comment:
Create custom modal view. For example `invalidKeySecretPair`:

```javascript
const invalidKeySecretPair = ({isOpen, closeModal, modalValues}) => (
    <ModalWindow
        isModalOpen={isOpen}
        onClose={closeModal}
        title={<FormattedMessage id='invalidKeySecretPair' values={modalValues} />}
        content={(
            <div>
                <button className="modal__button btn" onClick={closeModal}>
                  {this.props.intl.messages['ok']}
                </button>
            </div>
        )}
    />
);

const allModals = {
    invalidKeySecretPair,
};

const renderGlobalInformModel = ({modalId, modalValues, ...props}) => {
    if (allModals.hasOwnProperty(modalId)) {
        return allModals[modalId]({...props, modalValues});
    }
    else {
        // Render default modal view.
    }
}
```

Reply:
Move code to the separated file into dashboard directory (for example `confirmDialog.js`) and use it inside of `dashboard/ApiKey.js`.